### PR TITLE
Vectorize gamma MLE fit and thread gamma.ppf for gridded transform

### PR DESCRIPTION
**Pull request by benmsanderson**
## Summary
Two independent speedups to the 3D gamma quantile-mapping path used for gridded precipitation output.

### 1. `fit_distribution_parameters_3d`
Replaced the per-gridpoint `for i in range(n_spatial): scipy.stats.gamma.fit(x, floc=0)` loop with a vectorized `_vectorized_gamma_mle` that runs Choi-Wette + Newton iterations on `log(k) - psi(k) = log(mean(x)) - mean(log(x))` across all gridpoints at once. This is the same closed-form MLE scipy uses internally, just batched.

- Synthetic gamma data: agrees with scipy per-gridpoint to `rtol=1e-6`.
- Real NorESM2-MM precipitation: agrees to ~6e-5 relative (well beneath the noise model's own uncertainty).

### 2. `apply_distribution_transform` (3D gamma path)
Wrapped `scipy.special.gammaincinv` (which underlies `stats.gamma.ppf` and releases the GIL) in a `ThreadPoolExecutor` that chunks the spatial axis. Output is bitwise-identical to `stats.gamma.ppf`.

- Default: `min(8, cpu_count())` threads.
- Overridable via `METEOR_GAMMA_PPF_THREADS` env variable (`=1` disables threading).
- Malformed values silently fall back so a typo in a batch script cannot crash METEOR at import time.

## Wins on `gen_gridded`
| N | Baseline | This PR (combined with the batched-PCs PR) | Speedup |
|---|---:|---:|---:|
| 1 | 171 s | 45 s | 3.8× |
| 5 | 416 s | 154 s | 2.7× |

Post-change, the fixed cost (imports, netCDF loads, pattern-scaling load ≈ 22 s) plus the actual `gammaincinv` arithmetic (~64 s at N=5 across 8 threads) dominate the remaining time.

## Correctness
- `test_vectorized_gamma_mle_matches_scipy` — synthetic gamma at 200 gridpoints × 500 samples, rtol=1e-6.
- `test_vectorized_gamma_mle_handles_invalid_columns` — all-zero / all-NaN / single-positive columns fall back to safe defaults with no NaN propagation.
- `test_threaded_gamma_ppf_3d_matches_scipy` — asserts `array_equal` against `stats.gamma.ppf` for a `(200, 5000)` grid.
- `test_threaded_gamma_ppf_3d_single_thread_bypass` — `n_threads=1` produces the same output as multi-threaded.
- `test_resolve_gamma_ppf_threads_env_var` — env var override + fallback semantics.

## Test plan
- [ ] `pytest tests/unit/test_precipitation_transform.py` all pass
- [ ] Manual: `METEOR_GAMMA_PPF_THREADS=1 python -m pytest tests/unit/test_precipitation_transform.py -k ppf` still passes

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/meteor/precipitation_transform.py [hunk 1]
```diff
@@ -13,9 +13,139 @@
...
+    Parameters
+    ----------
+    u : ndarray, shape (..., n_spatial)
+        Uniform-scale quantiles (output of ``norm.cdf``).
+    shape_flat, scale_flat : ndarray, shape (n_spatial,)
+    n_threads : int, optional
+        Explicit thread count. If ``None`` (default), consults the
+        ``METEOR_GAMMA_PPF_THREADS`` env var, falling back to
+        ``min(8, cpu_count())``. Bypasses threading for small problems where
+        thread setup would dominate.
```
Comment (maritsandstad) [inline src/meteor/precipitation_transform.py:65]: If returns diminish above n_threads = 8, should a user choice much beyond this be overridden/yield a warning?

File: src/meteor/precipitation_transform.py [hunk 2]
```diff
@@ -13,9 +13,139 @@
...
+
+
+def _resolve_gamma_ppf_threads():
+    """Pick thread count for :func:`_threaded_gamma_ppf_3d`.
+
+    Reads the ``METEOR_GAMMA_PPF_THREADS`` environment variable so HPC users
+    can pin threads without touching Python (same pattern as ``OMP_NUM_THREADS``).
+    Falls back to ``min(8, cpu_count())`` — bounded because the underlying
+    scipy special function has diminishing returns past ~8 threads on typical
+    hardware. Set to ``1`` to disable threading entirely.
```
Comment (maritsandstad) [inline src/meteor/precipitation_transform.py:36]: If returns diminish much past 8 CPUs, should values way beyond this for METEOR_GAMMA_PPF_THREADS be capped? Warned about? Maybe no big deal and all users who set this can be expected to not send nonsense?

File: src/meteor/precipitation_transform.py [hunk 3]
```diff
@@ -13,9 +13,139 @@
...
+    shape, scale : ndarray (n_series,)
+    """
+    x = np.asarray(data, dtype=np.float64)
+
+    valid = (x > 0) & np.isfinite(x)
+    n_valid = valid.sum(axis=0)
+
+    # Sum and log-sum with invalid entries zeroed out
+    x_masked = np.where(valid, x, 0.0)
+    logx_masked = np.where(valid, np.log(np.maximum(x, np.finfo(float).tiny)), 0.0)
```
Comment (maritsandstad) [inline src/meteor/precipitation_transform.py:118]: Should this mask over x_masked and not x, so as to avoid crashes for logs of negative numbers in the evaluation?

File: tests/unit/test_precipitation_transform.py
Comment (maritsandstad) [inline tests/unit/test_precipitation_transform.py:1]: Add tests with negative values?

**Review by maritsandstad — COMMENTED** [review]
Left a few comments with questions here. Otherwise it looks alright to me

**Comment by benmsanderson** [comment]
Thanks @maritsandstad , good comments - fixes as follows:

**Capping threads past 8:**  returns flatten, but they dont peak.   Tests on 10 cores, 300×20000: 4.75× at 8 threads, 5.56× at 16, 5.86× at 256 — bitwise identical output throughout.  But - we submit one task per chunk, so METEOR_GAMMA_PPF_THREADS=100000 tries to spawn 100000 threads and dies. So: bounded rather than capped. Anything past 8× cpu_count gets clamped with a warning, and threads are also bounded by gridpoint count.

**Masking over x_masked:**  Switched to np.log(np.where(valid, x, 1.0)) — gives exactly 0.0 at invalid entries 

**Tests with negatives:**  Added test_vectorized_gamma_mle_excludes_negatives_without_warning covering negative/-inf/NaN/all-negative, asserting against scipy-on-the-survivors rather than just "is finite". Confirmed fails on a log-then-mask version, which returns the same numbers but emits warnings.

**Review by maritsandstad — APPROVED** [review]
